this is a refactor pull request

davidlougheed opened the pull request:
fixes #4

davidlougheed committed:
this is a refactor commit message

le-potate commented on Main.py:
```diff
@@ -2,431 +2,368 @@
+        if reg_vals:
+            new_reg_avg = int(statistics.mean(reg_vals))
+            old_reg_avg = avgs_list[0][0] if avgs_list else None
+            avg_diff = f"{str(new_reg_avg - old_reg_avg):+d}" if old_reg_avg is not None and day != "Sat" else None
```
Unknown format code 'd' for object of type 'str'

davidlougheed replied:
19e11bd1e01522040608edc461f68e8d7edf8cc6

le-potate commented on Main.py:
````diff
@@ -2,431 +2,368 @@
+        times_list = c.execute("SELECT Score,Date FROM Scores WHERE ID=? ORDER BY Date DESC",
+                               (ctx.author.id,)).fetchall()
+        if not times_list:
+            await ctx.bot.say('```No times found.```')
````
Should be `ctx.send`

davidlougheed replied:
19e11bd1e01522040608edc461f68e8d7edf8cc6

le-potate commented on Main.py:
```diff
@@ -2,431 +2,368 @@
+
+@bot.command()
+@commands.has_role(CROSSWORD_ROLE)
+async def deltime(ctx):
```
average isn't updated when a score is deleted

davidlougheed replied:
19e11bd1e01522040608edc461f68e8d7edf8cc6

le-potate replied:
The average still isn't updated when a score is deleted

le-potate replied:
Resolved in a0ffe74

le-potate commented on Main.py:
```diff
@@ -2,431 +2,368 @@
-@asyncio.coroutine
-async def update(self):
+async def on_ready():
+    print(f'Logged in as {bot.user.name} ({bot.user.id})')
```
I would also put this in the log

davidlougheed replied:
19e11bd1e01522040608edc461f68e8d7edf8cc6

le-potate commented on Main.py:
```diff
@@ -2,431 +2,368 @@
 
 # Logging configuration
 logger = logging.getLogger('discord')
 logger.setLevel(logging.ERROR)
-handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode = 'w')
+handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode='w')
 handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
 logger.addHandler(handler)
```
Nothing ends up in discord.log, every error message is in the console

davidlougheed replied:
19e11bd1e01522040608edc461f68e8d7edf8cc6

davidlougheed committed:
fix issues from potate's review
handle tracebacks from commands
factor out avg calculation, add to deltime (fixes #4)

davidlougheed committed:
formatting fix

davidlougheed committed:
potate please forgive me :(

le-potate commented on Main.py:
```diff
@@ -2,431 +2,409 @@
+    new_reg_avg: Optional[int] = None
+    new_sat_avg: Optional[int] = None
+
+    # A regular average can only be set if regular crossword scores exist
```
Average should be deleted if there's no score (if someone has one score and deletes it, the avg will stay the same because of this)

le-potate replied:
Resolved in 20d3d11

le-potate commented on Main.py:
```diff
@@ -2,431 +2,409 @@
+logger.setLevel(logging.INFO)
+handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode='w')
 handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
 logger.addHandler(handler)
```
Everytime the bot is restarted the log is overwritten

le-potate replied:
Resolved in 2f366f6

davidlougheed committed:
Append to log instead of overwriting

davidlougheed committed:
Delete average record if no scores left
